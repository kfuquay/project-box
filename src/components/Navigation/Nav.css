nav {
  background-color: #333;
  height: 5%;
  display: flex;
  flex-wrap: nowrap;
  justify-content: flex-end;
  align-items: baseline;
  flex-direction: row;
}

h1 {
  flex: 2 2 40%;
}

.nav-link {
  text-decoration: none;
  color: whitesmoke;
  margin: 0 8px;
  text-transform: uppercase;
  letter-spacing: 0.09em;
  font-weight: 300;
  font-size: 14px;
}

@media only screen and (max-width: 370px) {
  .nav-link {
    font-size: 12px;
    margin: 6px;
  }
}