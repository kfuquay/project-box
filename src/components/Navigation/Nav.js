import React, { Component } from "react";
import TokenService from "../../services/token-service";
import { NavLink } from "react-router-dom";
import "./Nav.css";
import ProjectContext from "../../Context/ProjectContext";

class Navigation extends Component {
  static contextType = ProjectContext;
  handleLogoutClick = () => {
    this.context.setCurrentUser("");
    TokenService.clearAuthToken();
  };

  renderLogoutLink() {
    return (
      <NavLink className="link" onClick={this.handleLogoutClick} to="/login">
        Logout
      </NavLink>
    );
  }

  renderLoginLink() {
    return (
      <NavLink to="/login" className="link">
        Log In
      </NavLink>
    );
  }
  render() {
    return (
      <nav role="navigation">
        <h1>
          <NavLink to="/" className="header link">
            Project Box
          </NavLink>
        </h1>
        <NavLink to="/home" className="link">
          Demo
        </NavLink>
        <NavLink to="/new" className="link">
          +Project
        </NavLink>
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </nav>
    );
  }
}

export default Navigation;
