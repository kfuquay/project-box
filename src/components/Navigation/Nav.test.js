import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Navigation from './Nav'

describe(`Navigation component`, () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Navigation />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
