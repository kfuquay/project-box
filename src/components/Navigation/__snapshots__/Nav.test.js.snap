// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders without crashing 1`] = `
<nav
  role="navigation"
>
  <h1>
    <NavLink
      className="header link"
      to="/"
    >
      Project Box
    </NavLink>
  </h1>
  <NavLink
    className="link"
    to="/home"
  >
    Demo
  </NavLink>
  <NavLink
    className="link"
    to="/new"
  >
    New Project
  </NavLink>
  <NavLink
    className="link"
    to="/login"
  >
    Log In
  </NavLink>
</nav>
`;
